packages/blocks/src/page-block/default/utils.ts

```diff
@@ -242,6 +252,31 @@ function binarySearchBlockEditingState(
     } else if (detectRect.bottom < y) {
       start = mid + 1;
     }
+
+    // if search failed, it may be caused by the mouse fall between two blocks
+    if (start > end) {
+      let targetIndex = -1;
+      // now start = end + 1, eg: [0, 1, ..., end start ..., blocks.length - 1]
+      if (start === blocks.length) {
+        targetIndex = end;
+      } else if (end === -1) {
+        targetIndex = start;
+      } else {
+        const { detectRect: prevDetectRect } = getBlockAndRect(blocks, end);
+        const { detectRect: nextDetectRect } = getBlockAndRect(blocks, start);
+        if (
+          Math.abs(prevDetectRect.bottom - y) < Math.abs(y - nextDetectRect.top)
+        ) {
```

Comment:
It can also be replaced with this: `y < prevDetectRect.bottom + (nextDetectRect.top - prevDetectRect.bottom) / 2`.

Reply:
Thanks! Updated, get rid of `Math.abs()` now (